src/main/webapp/app/exercises/text/participate/text-result/text-result-block.ts

```diff
@@ -29,7 +29,7 @@ export class TextResultBlock {
     }
 
     get feedbackType(): FeedbackType {
-        if (!this.feedback || !this.feedback.credits) {
+        if (!this.feedback || this.feedback.credits == null || Feedback.isEmpty(this.feedback)) {
```

Comment:
always issues with "0" evaluating to false 😄

Reply:
😄

Reply:
should we not check for undefined instead of null?

Reply:
```suggestion
        if (!this.feedback || this.feedback.credits == undefined || Feedback.isEmpty(this.feedback)) {
```

Reply:
I know it does not really make a difference, but we would like to avoid null and focus on undefined so I think it's better to use it

Reply:
a == null checks for null and undefined. I think the check is better because it fends against more errors.

Reply:
I would also support @kloessst, see here:
https://stackoverflow.com/questions/28975896/is-there-a-way-to-check-for-both-null-and-undefined

Checking only for undefined should be then `===` instead of your suggested `==` @krusche

Reply:
I think `== undefined` is logically the same as `== null`
Look at my proposal, I do not propose `=== undefined` which would be different.
We basically want to consistently use `undefined` instead of `null` in the client code (if possible), so I would prefer to not use `null` at all (if possible)

Reply:
@krusche jope, you are right. my bad! 
See: https://dorey.github.io/JavaScript-Equality-Table/